code/datums/components/subtype_picker.dm

```diff
@@ -0,0 +1,88 @@
+/**
+ * subtype picker component allows for an item to transform into its subtypes (this is not enforced and you can turn in whatever types, but
+ * i used this name as it was incredibly accurate for current usage of the behavior)
+ *
+ * Used for the null rod to pick the other holy weapons.
+ */
+/datum/component/subtype_picker
+	///on Attach, it's a list of types and their menu descriptions. after building, it's a list of lists with all the data each item needs to give to the radial menu.
+	var/list/subtype2descriptions
+	///list given to the radial menu to display
+	var/list/built_radial_list
+	///the radial will return a name of the wanted subtype, this is a list of references back to the type
+	var/list/name2subtype
```

Comment:
References is a weird way to describe this, makes me think it's an object

Reply:
i think at one point when it was an element it may have been. either way incorrect doc